ApplicationLibCode/ProjectDataModel/GeoMech/RimGeoMechPartCollection.h

```diff
@@ -0,0 +1,46 @@
+/////////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (C) 2021-     Equinor ASA
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include "cafPdmChildArrayField.h"
+#include "cafPdmField.h"
+#include "cafPdmObject.h"
+
+class RimGeoMechPart;
+class RimGeoMechCase;
+
+class RimGeoMechPartCollection : public caf::PdmObject
+{
+    CAF_PDM_HEADER_INIT;
+
+public:
+    RimGeoMechPartCollection();
+    ~RimGeoMechPartCollection() override;
+
+    void syncWithCase( RimGeoMechCase* geoCase );
+
+    bool shouldBeVisibleInTree() const;
```

Comment:
Rename and harmonize with `RimStreamlineInViewCollection::shouldBeAvailable`. Suggested name: `isAvaialable()` Looks more like the function `isPartEnabled()`